### ISSUE TITLE
Update default emulator model set

### ISSUE THREAD

**Issue by radka-j** [alan-turing-institute/autoemulate#843]
I'd propose we limit the default emulator set to just PyTorch models. If the users are interested in also training the sklearn models, they have to choose to do this explicitly.

**Comment by radka-j** [comment]
We also need to revisit the default set of GP classes used in AutoEmulate.

At the moment, AutoEmulate fits the main GP class as well as the 3 subclasses (rbf, matern 3/2 and matern 5/2). This is clearly a duplication of effort since the main GP class might also tune these 3 kernels.

To resolve this, I think we have 3 options for the default behaviour:
1. fit the generic GP and leave out the subclasses 
    - I am not a big fan of this because we could miss out the rbf or matern kernels simply due to the random search element
2. fit the 3 GP subclasses and leave out the generic one     
    - I prefer this to the first option as we are guaranteed to fit the most popular GP kernels which seems like a reasonable default behaviour
3. use all but remove the rbf and the matern kernels from the tuning params of the generic GP class
    - the disadvantage here is that this is quite a complicated behaviour to explain/document
    - what this would give us is essentially the option to fit the 3 key GP kernels + some "wild cards" which are covered in the hyperparameter tuning of the generic GP class

Related to this question is #844 - currently I am inclined not to add correlated GP subclasses and just have the one class. We could consider taking correlated GP out of the default emulator set though, I am not sure we have found a lot of use for correlated outputs. I suspect that capturing correlations is most important for spatio-temporal data which we will be handling explicitly with spatio-temporal models or not at all. But I dont have a very strong opinion on this.

@sgreenbury what do yo think?

**Comment by radka-j** [comment]
Note that the advantage of using the GP subclasses directly is that it also automatically resolves #534.

In some sense I see the GP kernel as akin to a different neural architecture. It would make sense to not treat that as a hyperparameter unless we have a more sophisticated method to search over compositions of kernels than random search (those methods exist but it might not be a high priority for us to implement them).
